pkg/pod/list.go

```diff
@@ -182,3 +183,79 @@ func WaitForAllPodsInNamespaceRunning(
 
 	return true, nil
 }
+
+// WaitForAllPodsInNamespacesHealthy waits until all pods in a list of namespaces that match options
+// are in healthy state.
+// An pod in a healthy state is in running phase and optionally in ready condition.
+//
+// nsNames lists the list of namespaces to monitor. Monitors all namespaces when empty.
+// timeout is the duration to wait for the pods to be healthy
+// includeSucceeded when true, implies that pods in succeeded phase are running.
+// checkReadiness when true, to also checks that the podConditions are ready.
+// ignoreFailedPods when true, to Ignore failed pods with restart policy set to never.
+// ignoreNamespaces is a list of namespaces to ignore.
+// options reduces the list of namespace to only the ones matching options.
+func WaitForAllPodsInNamespacesHealthy(
+	apiClient *clients.Settings,
+	nsNames []string,
+	timeout time.Duration,
+	includeSucceeded bool,
+	checkReadiness bool,
+	ignoreFailedPods bool,
+	ignoreNamespaces []string,
+	options ...metav1.ListOptions,
+) error {
+	logMessage := fmt.Sprintf("Waiting for all pods in %v namespaces", nsNames)
+	passedOptions := metav1.ListOptions{}
+
+	if len(options) > 1 {
+		glog.V(100).Infof("'options' parameter must be empty or single-valued")
+
+		return fmt.Errorf("error: more than one ListOptions was passed")
+	}
+
```

Comment:
Should there also be a `nil` check for `apiClient` before the calls to `List` etc?

Reply:
fixed